fragdenstaat_de/fds_cms/templates/foirequest/cms_plugins/map.html

```diff
@@ -0,0 +1,311 @@
+    <br>
+    <i style="background:#dc3545"></i> {% trans "Refused" %}
+    <br>
+    <i style="background:#17a2b8"></i> {% trans "Partially successful" %}
+    <br>
+    <i style="background:#fd7e14"></i> {% trans "Overdue" %}
+    <br>
+    <i style="background:#6c757d"></i> {% trans "Asleep" %}
+    <br>
+    {% if requests_without_geo > 0 %}
+        <hr class="legend-divider">
+        <div class="legend-note">
+            <i class="fas fa-info-circle"></i> {% blocktrans count counter=requests_without_geo %}{{ counter }} request without geo data{% plural %}{{ counter }} requests without geo data{% endblocktrans %}
+        </div>
+    {% endif %}
+</div>
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+        initMap();
+    });
+
+    // Cache for status icons
+    const statusIconCache = {};
+
+    function getStatusIcon(status) {
+        if (statusIconCache[status]) {
+            return statusIconCache[status];
+        }
+
+        const statusColors = {
+            'pending': '#ffc107',
+            'resolved': '#28a745',
+            'refused': '#dc3545',
+            'partially_successful': '#17a2b8',
+            'overdue': '#fd7e14',
+            'asleep': '#6c757d'
+        };
+
+        const color = statusColors[status] || '#6c757d';
+        const icon = L.divIcon({
+            className: 'custom-div-icon',
+            html: `<div style="background-color: ${color}; width: 12px; height: 12px; border-radius: 50%; border: 2px solid white; box-shadow: 0 0 4px rgba(0,0,0,0.3);"></div>`,
+            iconSize: [16, 16],
+            iconAnchor: [8, 8]
+        });
+
+        statusIconCache[status] = icon;
+        return icon;
+    }
+
+    function isValidCoordinates(coords) {
+        return coords && 
+               Array.isArray(coords) && 
+               coords.length === 2 && 
+               !isNaN(coords[0]) && 
+               !isNaN(coords[1]) &&
+               coords[0] >= -90 && 
+               coords[0] <= 90 &&
+               coords[1] >= -180 && 
+               coords[1] <= 180;
+    }
+
+    function createPopupContent(requests) {
+        let popupHtml = '<div class="foirequest-popup">';
+        requests.forEach((request, index) => {
+            popupHtml += `
+                <h4><a href="${request.url}">${request.title}</a></h4>
+                <p>${request.public_body_name}</p>
+                <p class="text-muted">${request.status_display}</p>
+                ${index < requests.length - 1 ? '<hr>' : ''}
+            `;
+        });
+        popupHtml += '</div>';
+        return popupHtml;
+    }
+
+    function initMap() {
+        // Initialize map with zoom controls
+        var map = L.map('map', {
+            zoomControl: true,
+            scrollWheelZoom: true
+        }).setView([51.1657, 10.4515], 6);
+
+        // Add OpenStreetMap tiles
+        L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
```

Comment:
Should probably use the cartodb light/dark style to be less noisy and emphasize the data on top.